ios/Exponent/Kernel/Views/Loading/EXManagedAppSplashScreenController.m

```diff
@@ -0,0 +1,92 @@
+//
+//  EXManagedAppSplashScreenController.m
+//  Expo Go (unversioned)
+//
+//  Created by andrew on 2021-06-07.
+//  Copyright © 2021 650 Industries. All rights reserved.
+//
+
+#import "EXManagedAppSplashScreenController.h"
+#import "MBProgressHUD.h"
+#import "EXSplashScreenHUDButton.h"
+
+
+@interface EXManagedAppSplashScreenController()
+
+@property (nonatomic, weak) NSTimer *warningTimer;
+@property (nonatomic, weak) MBProgressHUD *warningHud;
+@property (nonatomic, strong) UIView *splashScreenView;
+
+@end
+
+@implementation EXManagedAppSplashScreenController
+
+- (instancetype)initWithRootView:(UIView *)rootView splashScreenView:(UIView *)splashScreenView
+{
+  self = [super initWithRootView:rootView splashScreenView:splashScreenView];
+  
+  if (self) {
+    _splashScreenView = splashScreenView;
+    _splashScreenView.userInteractionEnabled = YES;
+  }
+  
+  return self;
+}
+
+- (void)showWithCallback:(void (^)(void))successCallback failureCallback:(void (^)(NSString * _Nonnull))failureCallback
+{
+  [super showWithCallback:^{
+    [self startSplashScreenVisibleTimer];
+    if (successCallback) {
+      successCallback();
+    }
+  } failureCallback:failureCallback];
+}
+
+- (void)hideWithCallback:(void (^)(BOOL))successCallback failureCallback:(void (^)(NSString * _Nonnull))failureCallback
+{
+  [super hideWithCallback:^(BOOL isSuccess){
+    if (self.warningTimer) {
+      [self.warningTimer invalidate];
+    }
+    
+    if (successCallback) {
+      successCallback(YES);
+    }
+  } failureCallback:failureCallback];
+}
+
+
+-(void)startSplashScreenVisibleTimer
+{
+  self.warningTimer = [NSTimer scheduledTimerWithTimeInterval:20.0
+                                                       target:self
+                                                     selector:@selector(showSplashScreenVisibleWarning)
+                                                     userInfo:nil
+                                                      repeats:NO];
+}
+
+- (void)showSplashScreenVisibleWarning
+{
+#if DEBUG
```

Comment:
Don't we want to behave the same way on Android? I've failed to find `if (debug)` there for `startSplashScreenWarningTimer` method in `ManagedAppSplashScreenController.kt` file  🤔